### PR TITLE
Automatic define form's name.

### PR DESCRIPTION
**Pull request by liverbool**
| Q | A |
| --- | --- |
| Bug Fix? | n |
| New Feature? | y |
| BC Breaks? | n |
| Deprecations? | n |
| Tests Pass? | n |
| Fixed Tickets |  |
| License | MIT |
| Doc PR |  |

This's just make `default form` same as `choice type` register. (See also #1927)

### CODE REVIEW

**Comment by pjedrzejewski** [comment]
Ha, smart idea, that would allow us to remove all ``getName`` methods. :) :+1: @liverbool

File: src/Sylius/Bundle/ResourceBundle/Form/Type/AbstractResourceType.php [hunk 1]
```diff
@@ -30,13 +30,20 @@
...
 
     /**
+     * @var string
+     */
+    protected $name;
+
+    /**
      * @param string   $dataClass        FQCN
      * @param string[] $validationGroups
+     * @param string $name
```
Comment (aRn0D) [inline src/Sylius/Bundle/ResourceBundle/Form/Type/AbstractResourceType.php]: wrong indentation

File: src/Sylius/Bundle/ResourceBundle/Form/Type/AbstractResourceType.php [hunk 2]
```diff
@@ -30,13 +30,20 @@
...
      */
-    public function __construct($dataClass, array $validationGroups = array())
+    public function __construct($dataClass, array $validationGroups = array(), $name = 'form')
```
Comment (aRn0D) [inline src/Sylius/Bundle/ResourceBundle/Form/Type/AbstractResourceType.php]: I ma not sure that the default value is a good idea

**Comment by aRn0D** [comment]
yep good idea !

File: src/Sylius/Bundle/ResourceBundle/Form/Type/AbstractResourceType.php (hunk 2 above)
Comment (pjedrzejewski) [inline src/Sylius/Bundle/ResourceBundle/Form/Type/AbstractResourceType.php]: Yeah.

Comment (liverbool) [inline src/Sylius/Bundle/ResourceBundle/Form/Type/AbstractResourceType.php]: Hmm, Any idea?

Comment (pjedrzejewski) [inline src/Sylius/Bundle/ResourceBundle/Form/Type/AbstractResourceType.php]: Maybe name should be first or second argument?

Comment (aRn0D) [inline src/Sylius/Bundle/ResourceBundle/Form/Type/AbstractResourceType.php]: @pjedrzejewski you're right, @liverbool i just want to remove it

Comment (liverbool) [inline src/Sylius/Bundle/ResourceBundle/Form/Type/AbstractResourceType.php]: I thinks `first` better and `choice` name should be first too.

Comment (pjedrzejewski) [inline src/Sylius/Bundle/ResourceBundle/Form/Type/AbstractResourceType.php]: Makes sense!

Comment (liverbool) [inline src/Sylius/Bundle/ResourceBundle/Form/Type/AbstractResourceType.php]: @pjedrzejewski  Are you confirm? :) We have a lot of files to change, i'll get start.

Comment (pjedrzejewski) [inline src/Sylius/Bundle/ResourceBundle/Form/Type/AbstractResourceType.php]: Yes, let's do this!

**Comment by aRn0D** [comment]
@liverbool ping

**Comment by aRn0D** [comment]
ping @liverbool

**Comment by liverbool** [comment]
@aRn0D **Very very sorry**, I have not free time to finish this PR, Feel free to close or create new PR.

:sob:

**Comment by pjedrzejewski** [comment]
@liverbool Maybe now? Just let us know if you will continue that or should we do this. Thanks a lot of the idea! :+1:

**Comment by liverbool** [comment]
@pjedrzejewski Sorry for my late, @sylius-team feel free to working on this feature.

;)

**Pull request closed without merging**